src/server/game/ArenaSpectator/ArenaSpectator.cpp

```diff
@@ -0,0 +1,191 @@
+                {
+                    switch (aura->GetSpellInfo()->Id)
+                    {
+                        case lfg::LFG_SPELL_DUNGEON_DESERTER:
+                        case lfg::LFG_SPELL_DUNGEON_COOLDOWN:
+                        case 26013: // bg deserter
+                        case 57724: // sated
+                        case 57723: // exhaustion
+                        case 25771: // forbearance
+                        case 15007: // resurrection sickness
+                        case 24755: // Tricked or Treated (z eventu)
+                            continue;
+                    }
+
+                    errors.push_back("Can't have negative auras.");
+                    break;
+                }
+
+        if (uint32 inviteInstanceId = player->GetPendingSpectatorInviteInstanceId())
+        {
+            if (Battleground* tbg = sBattlegroundMgr->GetBattleground(inviteInstanceId))
+                tbg->RemoveToBeTeleported(player->GetGUID());
+            player->SetPendingSpectatorInviteInstanceId(0);
+        }
+
+        bool bgPreparation = false;
+        if ((!handler->GetSession()->GetSecurity() && bgmap->GetBG()->GetStatus() != STATUS_IN_PROGRESS) ||
+            (handler->GetSession()->GetSecurity() && bgmap->GetBG()->GetStatus() != STATUS_WAIT_JOIN && bgmap->GetBG()->GetStatus() != STATUS_IN_PROGRESS))
+        {
+            bgPreparation = true;
+            handler->SendSysMessage("Arena is not in progress yet. You will be invited as soon as it starts.");
+            bgmap->GetBG()->AddToBeTeleported(player->GetGUID(), spectate->GetGUID());
+            player->SetPendingSpectatorInviteInstanceId(spectate->GetBattlegroundId());
+        }
+
+        if (!errors.empty())
+        {
+            handler->PSendSysMessage("To spectate, please fix the following:");
+            for (std::list<std::string>::const_iterator itr = errors.begin(); itr != errors.end(); ++itr)
+                handler->PSendSysMessage("- %s", (*itr).c_str());
+
+            return true;
+        }
+
+        if (bgPreparation)
+            return true;
+
+        player->SetPendingSpectatorForBG(spectate->GetBattlegroundId());
+        player->SetBattlegroundId(spectate->GetBattlegroundId(), spectate->GetBattlegroundTypeId(), PLAYER_MAX_BATTLEGROUND_QUEUES, false, false, TEAM_NEUTRAL);
+        player->SetEntryPoint();
+        float z = spectate->GetMapId() == 618 ? std::max(28.27f, spectate->GetPositionZ() + 0.25f) : spectate->GetPositionZ() + 0.25f;
+        player->TeleportTo(spectate->GetMapId(), spectate->GetPositionX(), spectate->GetPositionY(), z, spectate->GetOrientation(), TELE_TO_GM_MODE);
+        return true;
+    }
+
+    bool HandleSpectatorWatchCommand(ChatHandler* handler, char const* args)
+    {
+        if (!*args)
+            return true;
+
+        Player* player = handler->GetSession()->GetPlayer();
+        if (!player->IsSpectator())
```

Comment:
Brackets

Reply:
I'm not going to modify that code for now, I'm just moving it from one place to another.